src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -2618,6 +2667,32 @@ void SystemNative_GetDomainSocketSizes(int32_t* pathOffset, int32_t* pathSize, i
     *addressSize = sizeof(domainSocket);
 }
 
+int32_t SystemNative_Disconnect(intptr_t socket)
+{
+    int fd = ToFileDescriptor(socket);
+    int err;
+
+#if defined(__linux__)
+    // On Linux, we can disconnect a socket by connecting to AF_UNSPEC.
+    // For TCP sockets, this causes an abortive close.
+
+    // We don't need to clear the address, only the sa_family will be used.
```

Comment:
I know we spoke about this, but I'd still prefer it if the struct were cleared, rather than relying on an implementation that the address will be ignored.

Reply:
I'll clear it.